src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -840,7 +840,26 @@ public void execute() throws GitException, InterruptedException {
 
                 args.add(fastForwardMode);
                 args.add(rev.name());
-                launchCommand(args);
+
+                /* See JENKINS-45228 */
+                /* Git merge requires authentication in LFS merges, plugin does not authenticate the git merge command */
+                String defaultRemote = null;
+                try {
+                    defaultRemote = getDefaultRemote();
+                } catch (GitException e) {
+                    /* Nothing to do, just keeping defaultRemote = null */
+                }
+
+                if (defaultRemote != null && !defaultRemote.isEmpty()) {
+                    String repoUrl = getRemoteUrl(defaultRemote);
+                    StandardCredentials cred = credentials.get(repoUrl);
+                    if (cred == null) cred = defaultCredentials;
+                    launchCommandWithCredentials(args, workspace, cred, repoUrl);
```

Comment:
@MarkEWaite Any thoughts on the `CheckoutCommand` in this regard - see [here](https://github.com/jenkinsci/git-client-plugin/blob/git-client-3.0.0/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L2681)? Shouldn't credentials then also be added there by default? (See relevant issues [JENKINS-56569](https://issues.jenkins-ci.org/browse/JENKINS-56569) and [JENKINS-59139](https://issues.jenkins-ci.org/browse/JENKINS-59139)).

Reply:
I think that rework of the CheckoutCommand should be a separate pull request.  It will need to resolve how to handle git LFS pull when using older CLI git versions, how to handle it with newer CLI git versions, and how to handle the absence of the git LFS pull option.